# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:** Sofia Posey       <br>
**Class:** Python for Data Analytics

In [3]:
# Libraries
import numpy as np
import pandas as pd


In [4]:
# Synthetic Thanksgiving Data
np.random.seed()    # set seed to 2310
n =  50   # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  47.0              4.0               2.0       2.6            0
1  38.0              4.0               2.0       2.2            1
2  79.0              2.0               1.0       2.6            0
3  42.0              4.0               2.0       2.4            1
4  58.0              4.0               1.0       2.1            1

In [5]:
# Use the `describe` function to get the basic summary statistics
Thanksgiving.describe()


Ages  Dinner_Servings  Dessert_Servings   Duration  Macy_Parade
count  50.000000        50.000000         50.000000  50.000000    50.000000
mean   40.040000         3.620000          1.680000   2.520000     0.580000
std    18.722419         0.567486          0.767716   0.653718     0.498569
min     5.000000         2.000000          0.000000   1.000000     0.000000
25%    29.250000         3.000000          1.000000   2.000000     0.000000
50%    42.000000         4.000000          2.000000   2.500000     1.000000
75%    50.750000         4.000000          2.000000   2.975000     1.000000
max    90.000000         4.000000          3.000000   4.200000     1.000000

## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will being using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>
**Answer:** <br>
The minimum age in this family is 5 years old and the maximum age is 90 years old. This suggests that in this family there are no babys or new borns because the youngest in the family is a toddler. At the same time there are also elders in the family possible grandparents.

2. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>
**Answer:** <br>
The mean of Dinner_Duration of the Thanksgiving dinner is 2.53 hours, also the standard deviation of this variable is 0.65 hours. This indicates that most dinners have a duration of around 2.5 hours and that there is not a big change between the variants of the dinners.

3. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>
**Answer:** <br>
If we use the first method (2.52) + 2*(0.65) = 3.82. This indicates that 4.8 hours would be an outlier for this variant because is over the 3.82 limit. In other words, one of these dinners lasted more than the usual.

5. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>
**Answer:** <br> The mean for Dinner_servings is 3.62, and the median is 4. This suggests that half of the people ate around 4 servings and that the lower outliers is pulling the mean lower. This suggest that only a few family members ate less the the majority.

6. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be and what could this tell us about dessert consumption overall?  <br>
**Answer:** <br>
The maximum value in Dessert_servings is 3. I believe we can assume that the number of dessert servings represents a type of guest that enjoys sweets, a rough guess would be that this number represents the amount of children present at the family dinner. Overall, we see that in most family dinners there is an average of 2 children present maximum 3.

8. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?  <br>
**Answer:** <br> If the standard deviation is small it means the data is heavily concentrated around the average. This suggests that the demographics of the dinnder table members are similar so the can consume similar amounts. For example, if all at the table are young adults between 20 to 25 it might suggest that they are cousins and friends. 

9. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>
**Answer:** <br> I believe the mean, the first quartile and the minimum helps us find the light eaters. If the minimum amount of servings is 2, the mean is 3.62 and Q1 starts at 3 the anyone with lower than 3 servings is considered a light eater.

10. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>
**Answer:** <br>

11. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis or would we have to include something else?  <br>
**Answer:** <br>

12. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
**Answer:** <br>

13. what does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>
**Answer:** <br>

14. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>
**Answer:** <br>

15. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
**Answer:** <br>
